sklearn/utils/estimator_checks.py

```diff
@@ -331,6 +332,44 @@ def _construct_instance(Estimator):
     return estimator
 
 
+def _get_xfail_checks(estimator):
+    """Get xfail_check from estimator"""
+    if isinstance(estimator, type):
+        # try to construct estimator instance, if it is unable to then
+        # return then ignore xfail tag
+        try:
+            estimator = _construct_instance(estimator)
+        except Exception:
+            return {}
+
+    if not hasattr(estimator, "_get_tags"):
+        return {}
+
+    return estimator._get_tags()['_xfail_checks'] or {}
+
+
+def _check_warns_on_fail(estimator, check, xfail_checks_tag):
```

Comment:
Do we really need to pass the estimator? It looks like the logic only affects the check.

Reply:
It was not needed.

To address https://github.com/scikit-learn/scikit-learn/pull/16963/files#r414661838 it would be needed now.